### ISSUE TITLE
Host name is ignored when creating a host for Packet provider

### ISSUE THREAD

**Issue by soumyalj** [rancher/rancher#8827]
**Rancher versions:**
rancher/server: Master

**Docker version: (`docker version`,`docker info` preferred)** 17.05.0-ce

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**Ubuntu

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**Packet

**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**Single node

**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**Cattle

**Steps to Reproduce:**
1. On a master setup add a host from Packet provider using UI or API providing a name 

**Results:**
The host is added with the name `localhost` ignoring the name provided

**Comment by tfiduccia** [comment]
**Comments:**
I tried this on 1.4 and it's also giving me a name of "localhost". I then went and ssh'ed into the packet machine and did a hostname and was returned with "tanipackethost3" which is the correct name.  So packet has it correct but now we are not returning that name.

Checked the post from rancher ui and we are passing the name in
![image](https://cloud.githubusercontent.com/assets/11514927/26329655/27754652-3efd-11e7-9e8b-830c87cec0c7.png)
